### PR TITLE
STY: Apply ruff/pyupgrade rule UP031

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
New in [ruff v0.4.2](https://github.com/astral-sh/ruff/releases/tag/v0.4.2).

### CODE REVIEW

**Comment by DimitriPapadopoulos** [comment]
Fixing the `ImportError: cannot import name 'get_cmap' from 'matplotlib.cm'` CI error would require an updated [niworkflows](https://github.com/nipreps/niworkflows) (see https://github.com/nipreps/niworkflows/pull/866).

File: .maint/paper_author_list.py [hunk 1]
```diff
@@ -46,23 +46,26 @@ def _aslist(inlist):
     ]
 
     print(
-        'Some people made commits, but are missing in .maint/ '
-        'files: %s.' % ', '.join(unmatched),
+        'Some people made commits, but are missing in .maint/ ' 'files: {}.'.format(
```
Comment (effigies) [inline .maint/paper_author_list.py:49]: Isn't there a rule to catch these joined strings?


```suggestion
        'Some people made commits, but are missing in .maint/ files: {}.'.format(
```

File: .maint/paper_author_list.py [hunk 2]
```diff
@@ -46,23 +46,26 @@ def _aslist(inlist):
...
-        '%s.'
-        % '; '.join(
-            [
-                '{} \\ :sup:`{}`\\ '.format(i['name'], idx)
-                for i, idx in zip(author_matches, aff_indexes, strict=False)
-            ]
+        '{}.'.format(
+            '; '.join(
+                [
+                    '{} \\ :sup:`{}`\\ '.format(i['name'], idx)
```
Comment (effigies) [inline .maint/paper_author_list.py:60]:
```suggestion
                    rf'{i["name"]} \ :sup:`{idx}`\ '
```

File: .maint/paper_author_list.py (hunk 1 above)
Comment (DimitriPapadopoulos) [reply .maint/paper_author_list.py:49]: :smile: That was an issue with black, and it looks like ruff isn't any better.

File: .maint/update_authors.py [hunk 3]
```diff
@@ -274,18 +274,20 @@ def _aslist(value):
...
-        '%s.'
-        % '; '.join(
-            [
-                '{} \\ :sup:`{}`\\ '.format(i['name'], idx)
-                for i, idx in zip(hits, aff_indexes, strict=False)
-            ]
+        '{}.'.format(
+            '; '.join(
+                [
+                    '{} \\ :sup:`{}`\\ '.format(i['name'], idx)
```
Comment (effigies) [inline .maint/update_authors.py:280]:
```suggestion
                    rf'{i["name"]} \ :sup:`{idx}`\ '
```

File: fmriprep/cli/parser.py [hunk 4]
```diff
@@ -881,10 +881,9 @@ def parse_args(args=None, namespace=None):
     if output_dir == bids_dir:
         parser.error(
             'The selected output folder is the same as the input BIDS folder. '
-            'Please modify the output path (suggestion: %s).'
-            % bids_dir
+            f'Please modify the output path (suggestion: {bids_dir}).'
             / 'derivatives'
-            / ('fmriprep-%s' % version.split('+')[0])
+            / ('fmriprep-{}'.format(version.split('+')[0]))
```
Comment (effigies) [inline fmriprep/cli/parser.py:886]:
```suggestion
```

File: fmriprep/cli/parser.py [hunk 5]
```diff
@@ -881,10 +881,9 @@ def parse_args(args=None, namespace=None):
     if output_dir == bids_dir:
         parser.error(
             'The selected output folder is the same as the input BIDS folder. '
-            'Please modify the output path (suggestion: %s).'
-            % bids_dir
+            f'Please modify the output path (suggestion: {bids_dir}).'
             / 'derivatives'
```
Comment (effigies) [inline fmriprep/cli/parser.py:885]:
```suggestion
            'Please modify the output path (suggestion: {}).'.format(
                bids_dir / 'derivatives' / f'fmriprep-{version.split("+")[0]}'
            )
```

File: fmriprep/cli/workflow.py [hunk 6]
```diff
@@ -173,9 +173,7 @@ def build_boilerplate(config_file, workflow):
     config.load(config_file)
     logs_path = config.execution.fmriprep_dir / 'logs'
     boilerplate = workflow.visit_desc()
-    citation_files = {
-        ext: logs_path / ('CITATION.%s' % ext) for ext in ('bib', 'tex', 'md', 'html')
-    }
+    citation_files = {ext: logs_path / (f'CITATION.{ext}') for ext in ('bib', 'tex', 'md', 'html')}
```
Comment (effigies) [inline fmriprep/cli/workflow.py:176]:
```suggestion
    citation_files = {ext: logs_path / f'CITATION.{ext}' for ext in ('bib', 'tex', 'md', 'html')}
```

**Review by effigies — COMMENTED** [review]
ISC001 should detect and fix them: https://docs.astral.sh/ruff/rules/single-line-implicit-string-concatenation/

Some more suggestions that can't be auto-fixed.

File: fmriprep/cli/parser.py (hunk 4 above)
Comment (DimitriPapadopoulos) [reply fmriprep/cli/parser.py:887]: Not sure about this one. >You don't want to delete this line, do you?

**Comment by DimitriPapadopoulos** [comment]
> ISC001 should detect and fix them: https://docs.astral.sh/ruff/rules/single-line-implicit-string-concatenation/

Indeed, but `ruff format` should refrain from creating them when unfolding strings. I'll to reproduce and create a bug report.

**Comment by DimitriPapadopoulos** [comment]
It's a known ruff issue: https://github.com/astral-sh/ruff/issues/9457

In the meantime, there's no better solution than explicitly running `ruff check --select ISC001` **after** `ruff format` — while the intended order is `check` before `format`.

File: fmriprep/cli/parser.py (hunk 4 above)
Comment (effigies) [reply fmriprep/cli/parser.py:887]: Yes, I bundled it into the previous suggestion.

**Review by effigies — APPROVED** [review]

**Pull request merged**